include/zephyr/logging/log_msg.h

```diff
@@ -198,6 +198,17 @@ enum z_log_msg_mode {
 				  Z_LOG_MSG2_ALIGNMENT), \
 			 sizeof(uint32_t))
 
+/*
+ * With Zephyr SDK 0.14.2, aarch64-zephyr-elf-gcc (10.3.0) fails to ensure $sp
+ * is below the active memory during message construction. As a result,
+ * interrupts happening in the middle of that process can end up smashing active
+ * data and causing a logging fault. Work around this by inserting a compiler
+ * barrier after the allocation and before any use to make sure GCC moves the
+ * stack pointer soon enough
+ */
+
+#define Z_LOG_ARM64_VLA_PROTECT() compiler_barrier()
```

Comment:
Since its only for arm64 what about:
`#define Z_LOG_ARM64_VLA_PROTECT() IF_ENABLED(CONFIG_ARM64, (copiler_barrier()))`

Reply:
I thought #46986 was on arm32? Even if not, I still wonder if this is a problem in general GCC code or is specific to arm64?